### PR TITLE
fix downloadable? to account for other file types

### PR DESCRIPTION
**Pull request by johrstrom**
fix downloadable? to account for other file types to work on #3026.

### CODE REVIEW

File: apps/dashboard/app/models/posix_file.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class PosixFile
 
   attr_reader :path, :stat
 
-  delegate :basename, :descend, :parent, :join, :to_s, :read, :write, :mkdir, :directory?, :realpath, :pipe?, :readable?, to: :path
+  delegate :basename, :descend, :parent, :join, :to_s, :read, :write, :mkdir, :directory?, :realpath, :pipe?, :readable?, :file?, to: :path
```
Comment (CSC-swesters) [inline apps/dashboard/app/models/posix_file.rb:6]: Delegation to `:pipe?` is probably not needed with this commit? I didn't find the symbol used elsewhere with a quick look.

File: apps/dashboard/app/models/posix_file.rb [hunk 2]
```diff
@@ -62,10 +62,14 @@ def to_h
       owner:        PosixFile.username_from_cache(stat.uid),
       mode:         stat.mode,
       dev:          stat.dev,
-      downloadable: !pipe? && readable?
+      downloadable: downloadable?
     }
   end
 
+  def downloadable?
+    file? && readable?
```
Comment (CSC-swesters) [inline apps/dashboard/app/models/posix_file.rb:70]: I didn't figure out how the download button works, but should `directory?` be involved in this check, to enable zipping and downloading a directory?

Comment (johrstrom) [reply apps/dashboard/app/models/posix_file.rb]: Good catch! Apparently the tests use the download button, so I didn't notice.

File: apps/dashboard/app/models/posix_file.rb (hunk 1 above)
Comment (johrstrom) [reply apps/dashboard/app/models/posix_file.rb]: Yea it was for the other fix. I've removed it.

File: apps/dashboard/app/models/posix_file.rb [hunk 3]
```diff
@@ -3,7 +3,8 @@ class PosixFile
 
   attr_reader :path, :stat
 
-  delegate :basename, :descend, :parent, :join, :to_s, :read, :write, :mkdir, :directory?, :realpath, :pipe?, :readable?, to: :path
+  delegate :basename, :descend, :parent, :join, :to_s, :read, :write, :mkdir, to: :path
+  delegate :directory?, :realpath, :readable?, :file?, to: :path
```
Comment (johrstrom) [inline apps/dashboard/app/models/posix_file.rb:7]: To be clear, this removes `pipe?` while adding `file?` and putting them on 2 lines.

**Comment by Oglopf** [comment]
This looks good but do we want to add a test for this? I didn't see anything for it in the `test/models/files_test.rb` file but wrote this and the change works as expected:
```ruby
  test "cannot download files that are not downloadable?" do
    Dir.mktmpdir do |dir|
      file = File.join(dir, 'foo.text')
      FileUtils.touch(file)
      # make sure file is not readable, a condition of PosixFile#downloadable?
      FileUtils.chmod(0333, file)

      refute PosixFile.new(file).downloadable?
    end
  end
```

Unless the test is happening somewhere else that I missed.

**Comment by CSC-swesters** [comment]
> ```ruby
>  test "cannot download files that are not downloadable?" do
> ```

@Oglopf Looks like a good test, but I'd change the title of it to `cannot download files that are not readable`, so that it's more descriptive.

**Comment by johrstrom** [comment]
There's this system test: https://github.com/OSC/ondemand/blob/b2614412fb565e6323e096ecb070b85098bbf993/apps/dashboard/test/system/files_test.rb#L618 that I can add to and add some unit tests too.

**Comment by Oglopf** [comment]
@johrstrom ah, I missed the systems test. I think that with a unit test in the `files_test.rb` would be good coverage for this. Makes me wonder if other `posix_file` coverage is lacking in testing.

**Review by Oglopf — APPROVED** [review]
Looks great!

**Pull request merged**